### PR TITLE
New winning condition: killing all silos

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Killing all silos will set a 10 minutes timer for xenos. If they fail to make another silo in that time, marines win. Only works before shuttle hijack. Silos are boosted to 2000 hp instead of 400, so xenos can have time to defend them from suicidal podders

## Why It's Good For The Game

Remove the possibility of 4-5 xenos delaying indefinitly against 30 marines. Makes camp fobing less likely as it left silos vulnerable to tadpole or pod flanks. Give marines a clear objective to push on, as destroyed silos are currently barely an issue for the hive. This will maybe reduce cade hugging a little.

Overall, i think this change will be good for the flow of the game, I hope it won't lead to a trend of suicidal podders winning the game though 

## Changelog
:cl:
add: New victory condition for marines : killing all silos. 10 minutes countdown for a hive without silo to collapse
balance: Silo hp up to 2000 from 400
fix: Hivemind collapse shipside are now correctly xenos minor
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 1]
```diff
@@ -1114,3 +1114,24 @@ to_chat will check for valid clients itself already so no need to double check f
...
+		if(D.siloless_hive_timer)
+			deltimer(D.siloless_hive_timer)
+			D.siloless_hive_timer = null
+		return
+
+	if(D.siloless_hive_timer)
+		return
+
+	var/timer_length = 30 SECONDS
+	xeno_message("<span class='xenoannounce'>A sudden tremor ripples through the hive... the last silo was destroyed! The hive will collapse in nothing is done</span>", 3, TRUE)
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1136]:
```suggestion
	xeno_message("<span class='xenoannounce'>A sudden tremor ripples through the hive... the last silo was destroyed! The hive will collapse if nothing is done!</span>", 3, TRUE)
```

**Comment by Tennessee116** [comment]
Surprisingly great idea, considering how it sounds. Will there be any additional victory conditions for the xenomorphs?

**Comment by Walarks** [comment]
About "xenos win if they hijack". I will make a PR today or tomorrow that allows xenos to choose whether they would like to end the round or do shipside after they hijack.

File: code/modules/xenomorph/xeno_structures.dm [hunk 2]
```diff
@@ -69,10 +71,15 @@
...
 	if(associated_hive)
 		UnregisterSignal(associated_hive, list(COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, COMSIG_HIVE_XENO_MOTHER_CHECK))
-		//Since resin silos are more important now, we need a better notification.
-		associated_hive.xeno_message("<span class='xenoannounce'>A resin silo has been destroyed at [AREACOORD_NO_Z(src)]!</span>", 2, FALSE, src, 'sound/voice/alien_help2.ogg')
-		associated_hive = null
-		notify_ghosts("\ A resin silo has been destroyed at [AREACOORD_NO_Z(src)]!", source = get_turf(src), action = NOTIFY_JUMP)
+		var/datum/game_mode/infestation/distress/distress_mode
+		if(isdistress(SSticker.mode))
+			distress_mode = SSticker.mode
+			if (!distress_mode?.round_stage == DISTRESS_DROPSHIP_CRASHING)//No need to notify the xenos shipside
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:77]: excess ?.

File: code/modules/shuttle/marine_dropship.dm [hunk 3]
```diff
@@ -622,6 +618,9 @@
 
 /obj/machinery/computer/shuttle/marine_dropship/proc/do_hijack(obj/docking_port/mobile/marine_dropship/crashing_dropship, obj/docking_port/stationary/marine_dropship/crash_target/crash_target, mob/living/carbon/xenomorph/user)
 	crashing_dropship.set_hijack_state(HIJACK_STATE_CRASHING)
+	if(isdistress(SSticker.mode))
+		var/datum/game_mode/infestation/distress/distress_mode = SSticker.mode
+		distress_mode.round_stage = DISTRESS_DROPSHIP_CRASHING
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:623]: Why not do this with a signal for COMSIG_GLOB_DROPSHIP_HIJACKED

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 4]
```diff
@@ -1114,3 +1114,25 @@ to_chat will check for valid clients itself already so no need to double check f
 /datum/hive_status/proc/update_tier_limits()
 	tier3_xeno_limit = max(length(xenos_by_tier[XENO_TIER_THREE]),FLOOR((length(xenos_by_tier[XENO_TIER_ZERO])+length(xenos_by_tier[XENO_TIER_ONE])+length(xenos_by_tier[XENO_TIER_TWO]))/3+1,1))
 	tier2_xeno_limit = max(length(xenos_by_tier[XENO_TIER_TWO]),length(xenos_by_tier[XENO_TIER_ZERO]) + length(xenos_by_tier[XENO_TIER_ONE])+1 - length(xenos_by_tier[XENO_TIER_THREE]))
+
+///Handles the timer when all silos are destroyed
+/datum/hive_status/proc/handle_silo_death_timer()
+
+/datum/hive_status/normal/handle_silo_death_timer()
+	if(!isdistress(SSticker?.mode))
+		return
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1123]: Why the ?.

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 5]
```diff
@@ -1114,3 +1114,25 @@ to_chat will check for valid clients itself already so no need to double check f
 /datum/hive_status/proc/update_tier_limits()
 	tier3_xeno_limit = max(length(xenos_by_tier[XENO_TIER_THREE]),FLOOR((length(xenos_by_tier[XENO_TIER_ZERO])+length(xenos_by_tier[XENO_TIER_ONE])+length(xenos_by_tier[XENO_TIER_TWO]))/3+1,1))
 	tier2_xeno_limit = max(length(xenos_by_tier[XENO_TIER_TWO]),length(xenos_by_tier[XENO_TIER_ZERO]) + length(xenos_by_tier[XENO_TIER_ONE])+1 - length(xenos_by_tier[XENO_TIER_THREE]))
+
+///Handles the timer when all silos are destroyed
+/datum/hive_status/proc/handle_silo_death_timer()
+
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1120]:
```suggestion
///Handles the timer when all silos are destroyed
/datum/hive_status/proc/handle_silo_death_timer()
	return
```

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 6]
```diff
@@ -1114,3 +1114,25 @@ to_chat will check for valid clients itself already so no need to double check f
...
+			D.siloless_hive_timer = null
+		return
+
+	if(D.siloless_hive_timer)
+		return
+
+	var/timer_length = 10 MINUTES
+	xeno_message("<span class='xenoannounce'>A sudden tremor ripples through the hive... the last silo was destroyed! The hive will collapse in nothing is done</span>", 3, TRUE)
+	message_admins("Silo less collapse started")
+	D.siloless_hive_timer = addtimer(CALLBACK(D, /datum/game_mode.proc/siloless_hive_collapse), timer_length, TIMER_STOPPABLE)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1138]: This bit needs cleanup

File: code/datums/gamemodes/distress.dm [hunk 7]
```diff
@@ -289,6 +290,24 @@
...
+	if(round_finished)
+		return
+	if(round_stage == DISTRESS_DROPSHIP_CRASHING)
+		return
+	round_finished = MODE_INFESTATION_M_MAJOR
+
+
+/datum/game_mode/infestation/distress/get_siloless_collapse_countdown()
+	if(!siloless_hive_timer)
+		return
```
Comment (TiviPlus) [inline code/datums/gamemodes/distress.dm:305]:
```suggestion
		return 0
```
?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm (hunk 4 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1123]: Cause copy pasta from hive collapse code

File: code/modules/shuttle/marine_dropship.dm (hunk 3 above)
Comment (BraveMole) [reply code/modules/shuttle/marine_dropship.dm:623]: But why? we are just changing a flag, no need for a signal

**Comment by neohenry** [comment]
> 
> 
> About "xenos win if they hijack". I will make a PR today or tomorrow that allows xenos to choose whether they would like to end the round or do shipside after they hijack.

Will refer your PR in this PR: https://github.com/tgstation/TerraGov-Marine-Corps/pull/5786

TRULY, THE XENO PRS THAT SHALL HAVE AN IMPACT IN DISTRESS!

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 8]
```diff
@@ -449,7 +449,7 @@
 
 
 /datum/hive_status/normal/check_ruler()
-	if(!SSticker?.mode || !(SSticker.mode.flags_round_type & MODE_XENO_RULER))
+	if(!SSticker.mode || !(SSticker.mode.flags_round_type & MODE_XENO_RULER))
 		return TRUE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:453]: use ?.

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 9]
```diff
@@ -1114,3 +1114,26 @@ to_chat will check for valid clients itself already so no need to double check f
 /datum/hive_status/proc/update_tier_limits()
 	tier3_xeno_limit = max(length(xenos_by_tier[XENO_TIER_THREE]),FLOOR((length(xenos_by_tier[XENO_TIER_ZERO])+length(xenos_by_tier[XENO_TIER_ONE])+length(xenos_by_tier[XENO_TIER_TWO]))/3+1,1))
 	tier2_xeno_limit = max(length(xenos_by_tier[XENO_TIER_TWO]),length(xenos_by_tier[XENO_TIER_ZERO]) + length(xenos_by_tier[XENO_TIER_ONE])+1 - length(xenos_by_tier[XENO_TIER_THREE]))
+
+///Handles the timer when all silos are destroyed
+/datum/hive_status/proc/handle_silo_death_timer()
+	return
+/datum/hive_status/normal/handle_silo_death_timer()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1121]:
```suggestion
/datum/hive_status/proc/handle_silo_death_timer()
	return

/datum/hive_status/normal/handle_silo_death_timer()
```

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 10]
```diff
@@ -1114,3 +1114,26 @@ to_chat will check for valid clients itself already so no need to double check f
...
+	if(GLOB.xeno_resin_silos.len)
+		if(D.siloless_hive_timer)
+			deltimer(D.siloless_hive_timer)
+			D.siloless_hive_timer = null
+		return
+
+	if(D.siloless_hive_timer)
+		return
+
+	message_admins("Silo less collapse started")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1135]: .

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 11]
```diff
@@ -1114,3 +1114,26 @@ to_chat will check for valid clients itself already so no need to double check f
...
+	xeno_message("<span class='xenoannounce'>A sudden tremor ripples through the hive... the last silo was destroyed! The hive will collapse in nothing is done</span>", 3, TRUE)
+	
+	var/timer_length = 10 MINUTES
+	D.siloless_hive_timer = addtimer(CALLBACK(D, /datum/game_mode.proc/siloless_hive_collapse), timer_length, TIMER_STOPPABLE)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1139]: why are you declaring the var here

File: code/modules/xenomorph/xeno_structures.dm [hunk 12]
```diff
@@ -46,6 +46,8 @@
 	number++
 
 	GLOB.xeno_resin_silos += src
+	if(associated_hive)
+		associated_hive.handle_silo_death_timer()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:50]: ?.

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by BraveMole** [comment]
Closing this, will come back with more features

**Pull request closed without merging**